chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+package co.chatsdk.ui;
+
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
+    }
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+
+        private WeakReference<ImageListAdapter> adapter;
+        public ImageView imageView;
+        public Uri thisImageUri;
+        public int thisImagePosition;
+
+        public MyViewHolder(ImageListAdapter la, ImageView v) {
+            super(v);
+            this.adapter = new WeakReference<ImageListAdapter>(la);
```

Comment:
Remove `<ImageListAdapter>`

Reply:
Unchecked assignment. I hope this is not a problem.